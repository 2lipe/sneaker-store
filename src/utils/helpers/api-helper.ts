import axios from 'axios';
import { baseConfig } from '../../config/baseConfig';

export const api = axios.create({
  baseURL: baseConfig.baseApiUrl,
});
