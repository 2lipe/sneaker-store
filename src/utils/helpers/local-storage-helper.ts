/* eslint-disable consistent-return */
import { ISneakers } from '../../models/Interfaces/ISneakers';

export const saveSneakersLocalStorage = (data: ISneakers[]) => {
  if (typeof window !== 'undefined') {
    window.localStorage.setItem('@trustly-store:sneakers', JSON.stringify(data));
  }
};

export const getSneakerLocalStorage = () => {
  if (typeof window !== 'undefined') {
    const sneakersString = window.localStorage.getItem('@trustly-store:sneakers');

    if (sneakersString) {
      const sneakers: ISneakers[] = JSON.parse(sneakersString);

      return sneakers;
    }

    return null;
  }
};
