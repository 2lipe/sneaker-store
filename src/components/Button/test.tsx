import React from 'react';

import { screen } from '@testing-library/react';
import { renderWithTheme } from '../../utils/tests/helpers';

import { Button } from '.';

describe('<Button />', () => {
  it('should be render the medium size by default', () => {
    const { container } = renderWithTheme(<Button>Add to cart</Button>);

    expect(screen.getByRole('button', { name: /Add to cart/i })).toHaveStyle({
      height: '4rem',
      width: '27rem',
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should be render the large size', () => {
    renderWithTheme(<Button size="large">Add to cart</Button>);

    expect(screen.getByRole('button', { name: /Add to cart/i })).toHaveStyle({
      height: '4rem',
      width: '30rem',
    });
  });
});
