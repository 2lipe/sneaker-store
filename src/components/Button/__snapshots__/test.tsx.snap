// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Button /> should be render the medium size by default 1'] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #6B8067;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  height: 4rem;
  width: 27rem;
  border-radius: 0.4rem;
}

.c0:hover {
  background-color: #556652;
}

.c0 span {
  color: #FFFFFF;
  font-size: 1.4rem;
  font-weight: 600;
}

<button
  class="c0"
>
  <span>
    Add to cart
  </span>
</button>
`;
